### PR TITLE
NEW Card component

### PR DESCRIPTION
**Pull request by ShimiSun**
#### What does this PR do?
Adds Card component to the component library.
I'm open to feedback, especially regarding the assignment of default values on the theme.

Design system Card https://www.figma.com/file/WzYBJncJEbjA2JU9GqZMZB/HPE-Cards?node-id=0%3A1
#### Where should the reviewer start?
doc.js for API
#### What testing has been done on this PR?
storybook + jest
#### How should this be manually tested?
storybook
#### Any background context you want to provide?
Moving away from aries-core
#### What are the relevant issues?
#3489 

#### Do the grommet docs need to be updated?
done.
#### Should this PR be mentioned in the release notes?
Of course. bumping up the 'minor' version.
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: src/js/themes/base.js [hunk 1]
```diff
@@ -511,6 +511,26 @@ export const generate = (baseSpacing = 24, scale = 6) => {
       },
       heading: { level: '4' }, // level ranges from 1-6
     },
+    // add to index.d.ts
```
Comment (ShimiSun) [inline src/js/themes/base.js:514]:
```suggestion

```

File: src/js/components/Card/Card.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import { Header } from '../Header';
+import { Footer } from '../Footer';
+
+export const Card = forwardRef(({ children, header, footer, ...rest }) => {
+  const theme = useContext(ThemeContext) || defaultProps.theme;
+
+  return (
+    <Box align="start" justify="between" {...theme.card} {...rest}>
+      {header && (
+        <Header
```
Comment (ericsoderberghp) [inline src/js/components/Card/Card.js:17]: Hmm... something feels a bit odd here. What if the caller wants to change the `justify` on the Header? And, passing multiple contents via something like `<Card header={[<Avatar />, <Button />]} />` seems heavy

I think we are wanting Card to have specific default Header and Footer styles but allow them to be customized where needed. 

What about treating Card more like Table?:
```
<Card>
  <CardHeader />
  <CardBody />
  <CardFooter />
 </Card>
```
Where CardBody is only needed if you have CardHeader or CardFooter.
Just thinking.

**Comment by erickriva** [comment]
I'm trying to develop some ready-to-use components using Grommet, one of them is a Card.
Maybe my implementation can help to think in different ways.

[Link to Codesandbox](https://codesandbox.io/s/grommet-test-4qgos?file=/src/components/general/Card/index.tsx)

**Comment by ShimiSun** [comment]
Great feedback!! I filed a new PR that adapts the suggested API. Thank you!

**Pull request closed without merging**